cypress/cypress/e2e/EngageSphere_API.cy.js

```diff
@@ -0,0 +1,62 @@
+// Recupera clientes com sucesso (por exemplo, verifica o código de status 200)
+describe('Recuperação de Clientes', () => {
+    it('Deve recuperar clientes com sucesso', () => {
+      cy.request('GET', 'http://localhost:3001/customers?page=2&limit=10&size=All')
+        .then((response) => {
+          expect(response.status).to.equal(200);
+        });
+    });
+  });
+
+// Pagina a lista de clientes corretamente
+
+describe('Teste de Listagem de Clientes', () => {
+    it('Deve retornar a lista de clientes com sucesso', () => {
+      cy.request('GET', 'http://localhost:3001/customers?page=1&limit=10&size=All').then((response) => {
+        expect(response.status).to.equal(200);
+
+        expect(response.body).to.have.all.keys('customers', 'pageInfo');
+        expect(response.body.customers).to.have.length.gt(0);
+        expect(response.body.pageInfo).to.have.all.keys('currentPage', 'totalPages', 'totalCustomers');
+      });
+    });
+  });
+
+//Filtra clientes por tamanho corretamente
+
+describe('Teste de filtro de clientes por tamanho', () => {
+    it('Deve filtrar clientes por tamanho corretamente', () => {
+      cy.request('GET', 'http://localhost:3001/customers?page=1&limit=10&size=Small').then((response) => {
+
+        expect(response.status).to.equal(200);
+        expect(response.body.customers).to.have.length.gt(0);
+        response.body.customers.forEach((cliente) => {
+          expect(getSize(cliente)).to.equal('Small');
+        });
+      });
+    });
+  });
+
+  const getSize = ({ employees }) => {
+    if (employees >= 50000) return 'Very Large Enterprise';
+    if (employees >= 10000) return 'Large Enterprise';
+    if (employees >= 1000) return 'Enterprise';
+    if (employees >= 100) return 'Medium';
+    return 'Small';
+  };
+
+ //Retorna a estrutura correta da resposta (ou seja, propriedades dos customers e pageInfo)
+
+ describe('Teste de Estrutura da Resposta', () => {
+    it('Deve retornar a estrutura correta da resposta', () => {
+      const url = 'http://localhost:3001/customers?page=1&limit=50&size=All';
```

Comment:
Recomendo mover a string `http://localhost:3001` para a `env` `API_URL` no arquivo de configurações do Cypress (`cypress.config.js`) e nos testes de API, no início do arquivo, definir uma variável do tipo `const` chamada `CUSTOMERS_API_URL`, a qual guarda o seguinte valor: `${Cypress.env('API_URL')}/customers`.

Dessa forma, tal variável pode ser interpolada com o resto da string, sem a necessidade de repetí-la em diferentes lugares.